### PR TITLE
Blob Store compress default to true

### PR DESCRIPTION
**Pull request by henningandersen**
Changed default of compress setting from false to true for blob store
repositories. This aligns the code with documentation and also seems
like the better default.

### CODE REVIEW

**Comment by henningandersen** [comment]
@elasticmachine run elasticsearch-ci/1

Failed due to #40030

File: docs/reference/migration/migrate_8_0/snapshots.asciidoc [hunk 1]
```diff
@@ -11,3 +11,14 @@ The `repositories.fs.compress` setting has been removed.
 
 Instead use the repository specific `compress` setting to enable compression. See <<modules-snapshots>> for information
 on the `compress` setting.
+
+[float]
+==== Compression of meta data files is now default on
+
+Previously, the default value for `compress` was `false`. The default has been changed to `true`. This aligns the
+default with documentation.
```
Comment (ywelsch) [inline docs/reference/migration/migrate_8_0/snapshots.asciidoc]: should we change the snapshot docs for 7.x and below to say that the default is false? We can then leave this sentence out.

**Review by ywelsch — CHANGES_REQUESTED** [review]

File: docs/reference/migration/migrate_8_0/snapshots.asciidoc [hunk 2]
```diff
@@ -11,3 +11,14 @@ The `repositories.fs.compress` setting has been removed.
 
 Instead use the repository specific `compress` setting to enable compression. See <<modules-snapshots>> for information
 on the `compress` setting.
+
+[float]
+==== Compression of meta data files is now default on
```
[the remark this answers is not in the document]
Comment (henningandersen) [reply docs/reference/migration/migrate_8_0/snapshots.asciidoc]: Fixed in fd7d5fed2a34293edf4c9a772f73d9570971dcb5

File: docs/reference/migration/migrate_8_0/snapshots.asciidoc (hunk 1 above)
Comment (henningandersen) [reply docs/reference/migration/migrate_8_0/snapshots.asciidoc]: Removed that sentence in fd7d5fed2a34293edf4c9a772f73d9570971dcb5 and will do a follow-up PR against 7.x

**Comment by henningandersen** [comment]
@ywelsch please notice the additional documentation changes, the compress option was documented on specific repository types too and here it was stated default false (now changed to true).

**Review by ywelsch — APPROVED** [review]
LGTM

**Pull request merged**